lib/std/Io/Writer.zig

```diff
@@ -2276,7 +2273,12 @@ pub const Discarding = struct {
         const d: *Discarding = @alignCast(@fieldParentPtr("writer", w));
         d.count += w.end;
         w.end = 0;
-        if (limit == .nothing) return 0;
```

Comment:
I don't think this is correct, the limit must still be enforced even if this is a discarding writer...

The `buffered_len != 0` case just below is also missing limit enforcement.